pkg/daemon/daemon.go

```diff
@@ -491,6 +491,15 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			}
 		}
 	}
+
+	d, r, err := trySetOvsHWOL(utils.SwitchdevDeviceExists(latestState))
```

Comment:
iiuc, we are adding 1 more reboot when we enable switchdev hw offload since that code path also uses this and in that case, we already have to have an early dev-link command on each boot. Is my understanding correct? In other words, we pay an extra reboot at the time we enable it to gain a way to get rid of it when we disable it. 

Thinking along the same lines, would it even make sense to have ovs set dev-link to switchdev mode? I'm not sure we want that, but then we could go back to one reboot.

Reply:
In that note. If we make the hw-offload config in OVS ephemeral then we can avoid this all together. Having an ephemeral config has it's pros and cons.

Reply:
Yes, we are currently paying an extra reboot when the daemon detects a device has been put into switchdev mode.

Reply:
One potential issue is that pods might become Ready with a VF while HWOL is turned off. I don't think there will be network connectivity for those pods.

Reply:
See the most recent update, we can instead manually restart ovs from the config daemon to avoid the extra restart.

With the new behavior, we will trigger configuring ovs-hwol after a node reboots due to a change in the node state policy. We will detect if the node has booted with or without switchdev devices and set ovs-hwol accordingly

Reply:
I'm ok with a restart of ovs. In theory, it should be possible to dynamically take care of this as well to be honest.